pandas/core/strings.py

```diff
@@ -1122,21 +1127,17 @@ def str_join(arr, sep):
     Example with a list that contains non-string elements.
 
     >>> s = pd.Series([['lion', 'elephant', 'zebra'],
-    ...                [1.1, 2.2, 3.3],
-    ...                ['cat', np.nan, 'dog'],
-    ...                ['cow', 4.5, 'goat']
-    ...                ['duck', ['swan', 'fish'], 'guppy']])
+    ...            [1.1, 2.2, 3.3],
+    ...            ['cat', np.nan, 'dog'],
+    ...            ['cow', 4.5, 'goat'],
+    ...            ['duck', ['swan', 'fish'], 'guppy']])
     >>> s
     0        [lion, elephant, zebra]
     1                [1.1, 2.2, 3.3]
     2                [cat, nan, dog]
     3               [cow, 4.5, goat]
     4    [duck, [swan, fish], guppy]
     dtype: object
-
-    Join all lists using an '-', the lists containing object(s) of types other
-    than str will become a NaN.
```

Comment:
I think this comment was useful, any particular reason to remove it?_

Reply:
We removed it after misdiagnosing an issue, I will reinstate it. Thank you!